src/transcript/keccak.rs

```diff
@@ -0,0 +1,126 @@
+use std::marker::PhantomData;
+use tiny_keccak::{Keccak, Hasher};
+use ark_ec::{AffineRepr, CurveGroup};
+use ark_ff::{BigInteger, Field, PrimeField};
+
+use crate::transcript::Transcript;
+
+/// KecccakTranscript implements the Transcript trait using the Keccak hash
+pub struct KeccakTranscript<C: CurveGroup> {
+    sponge: Keccak,
+    phantom: PhantomData<C>,
+}
+
+#[derive(Debug)]
+pub struct KeccakConfig {}
+
+impl<C: CurveGroup> Transcript<C> for KeccakTranscript<C> {
+    type TranscriptConfig = KeccakConfig;
+    fn new(config: &Self::TranscriptConfig) -> Self {
+        let _ = config;
+        let sponge = Keccak::v256();
+        Self {
+            sponge,
+            phantom: PhantomData,
+        }
+    }
+
+    fn absorb(&mut self, v: &C::ScalarField) {
+        self.sponge.update(&(v.into_bigint().to_bytes_le()));
+    }
+    fn absorb_vec(&mut self, v: &[C::ScalarField]) {
+        for _v in v {
+            self.sponge.update(&(_v.into_bigint().to_bytes_le()));
+        }
+    }
+    fn absorb_point(&mut self, p: &C) {
+        self.sponge.update(&prepare_point(p))
+    }
+    fn get_challenge(&mut self) -> C::ScalarField {
+        let mut output = [0u8; 32];
+        self.sponge.clone().finalize(&mut output);
+        self.sponge.update(&[output[0]]);
+        C::ScalarField::from_le_bytes_mod_order(&[output[0]])
+    }
+    fn get_challenge_nbits(&mut self, nbits: usize) -> Vec<bool> {
+        // TODO
+        vec![]
+    }
+    fn get_challenges(&mut self, n: usize) -> Vec<C::ScalarField> {
+        let mut output = [0u8; 32];
+        self.sponge.clone().finalize(&mut output);
+        self.sponge.update(&[output[0]]);
+
+        let c: Vec<C::ScalarField> = output
+            .iter()
+            .map(|c| C::ScalarField::from_le_bytes_mod_order(&[*c]))
+            .collect();
+        c[..n].to_vec()
+    }
+}
+
+// Returns the point coordinates in Fr, so it can be absrobed by the transcript. It does not work
+// over bytes in order to have a logic that can be reproduced in-circuit.
+fn prepare_point<C: CurveGroup>(p: &C) -> Vec<u8> {
```

Comment:
This method was done in [transcript/poseidon.rs](https://github.com/privacy-scaling-explorations/folding-schemes/blob/8256c27609a2effbf9a255f0709eb0b1c3055b36/src/transcript/poseidon.rs#L57) to have a compatible logic between native and in-circuit, being mindful about the finite fields and number of constraints.
Since the KeccakTranscript is not planned to be used in-circuit, we don't need to take care about the field elements and we can directly serialize the curve point as byte array.
How do you see just removing this method, and in the `absorb_point`(line 37) just serialize the curve point into bytes?

Reply:
You mean like this? This works as well.
I made sure that it creates another bytes when we put another curve points (in local).

```rust
    fn absorb_point(&mut self, p: &C) {
        let mut serialized = vec![];
        p.serialize_compressed(&mut serialized).unwrap();
        self.sponge.update(&(serialized))
    }
```

✅ https://github.com/privacy-scaling-explorations/folding-schemes/pull/27/commits/bc7b58c1e4cf5c86cde0d178bfa46764ea2baca4

Reply:
Yes, perfect!